OpenRA.Mods.Common/Traits/Player/EnemyWatcher.cs

```diff
@@ -85,11 +84,28 @@ public void Tick(Actor self)
 				if (lastKnownActorIds.Contains(actor.Actor.ActorID))
 					continue;
 
+				var notificationPlayed = playedNotifications.Contains(actor.Trait.Info.Notification);
```

Comment:
I could think about moving this to a new player-level announcement trait if people are against it, but I consider that an overkill.
I also consider the requested on IRC rearchitecturing a huge overkill.